node/src/prover/router.rs

```diff
@@ -45,7 +45,8 @@ impl<N: Network, C: ConsensusStorage<N>> Handshake for Prover<N, C> {
         let conn_side = connection.side();
         let stream = self.borrow_stream(&mut connection);
         let genesis_header = *self.genesis.header();
-        self.router.handshake(peer_addr, stream, conn_side, genesis_header).await?;
+        let restrictions_id = Field::zero(); // Provers may bypass restrictions, since they do not validate transactions.
+        self.router.handshake(peer_addr, stream, conn_side, genesis_header, restrictions_id).await?;
```

Comment:
I feel like the handshake method should take an option, then no? Why send extra data if it's unnecessary?

Reply:
I agree it would have been cleaner, if the PR was not merged yet or we were not in a crunch, I would have pushed the change. Thx for the suggestion